src/modules/rewards/services/op-rebate-service.ts

```diff
@@ -337,6 +337,17 @@ export class OpRebateService {
     }
   }
 
+  private getInputTokenPrice(deposit: PartialDeposit): string {
+    if (
+      deposit.sourceChainId === ChainIds.blast &&
+      deposit.token.symbol === "USDB" &&
+      deposit.outputToken.symbol === "DAI"
+    ) {
+      return deposit.outputTokenPrice.usd;
+    }
+    return deposit.price.usd;
+  }
+
```

Comment:
This is a clone - can we not create a parent service with this logic & have both the OP/Arb classes inherit from it?

Reply:
cc @amateima as this could be an interesting refactoring opportunity for the next tech debt cycle 😄

Reply:
I think it would make sense to do a refactor when we are going to add another rebate service in the future.